salt/docker/init.sls

```diff
@@ -16,6 +16,7 @@ include:
         Environment="NO_PROXY={{ salt['pillar.get']('proxy:no_proxy', '') }}"
   cmd.run:
     - name: systemctl daemon-reload
+    - stateful: True
```

Comment:
Salt will always assume a `cmd.run` will "change something", as it cannot know if something has changed when running that command. With `stateful: True` you acknowledge to follow their protocol, so salt will be able to know if a `cmd.run` did change anything in the system state or not.

Even with `stateful: True` and empty `stdout` that doesn't hide the fact that the `file.managed` `/etc/systemd/system/docker.service.d/proxy.conf` did actually change, so if you watch for `/etc/systemd/system/docker.service.d/proxy.conf` or `file: /etc/systemd/system/docker.service.d/proxy.conf` there will be changes, because at least one of the actions did change (in this case if contents change with `file.contents`).